testimages/linux/install-java.sh

```diff
@@ -7,9 +7,23 @@ if [ -z "$AGENT_IMAGE" ]; then
   exit 1
 fi
 
-sudo apt-get update && sudo apt-get install openjdk-17-jre-headless -y && java -version
+install_java() {
+  export DEBIAN_FRONTEND=noninteractive
+  apt-get update
+  apt-get install -y wget apt-transport-https gpg
+  wget -qO - https://packages.adoptium.net/artifactory/api/gpg/key/public | gpg --dearmor | tee /etc/apt/trusted.gpg.d/adoptium.gpg > /dev/null
+  echo "deb https://packages.adoptium.net/artifactory/deb $(awk -F= '/^VERSION_CODENAME/{print$2}' /etc/os-release) main" | tee /etc/apt/sources.list.d/adoptium.list
+  apt-get update
+  # jre is sufficient in the integration tests (jdk would be required if we are building a java projects inside the agent.)
+  apt-get install -y temurin-21-jre
```

Comment:
`headless-jre` might suffice, if that is offered.

Reply:
`headless-jre` isn't there in this distribution. not sure if should have stick with openjdk which has headless
(but jenkins docker-agent uses adoptium so thought to use this).

ig, will not change again now. maybe some other time will see